### PR TITLE
Nnls fit

### PR DESCRIPTION
**Pull request by licode**
There are some minor corrections on nnls fitting, so that users have more controls on how to define weights.

### CODE REVIEW

File: skxray/core/fitting/xrf_model.py [hunk 1]
```diff
@@ -1064,72 +1064,39 @@ def nnls_fit(spectrum, expected_matrix):
...
+    nnls is chosen as amplitude of each element should not be negative.
     """
-    experiments = spectrum
-    standard = expected_matrix
 
-    [results, residue] = nnls(standard, experiments)
-    return results, residue
-
-
-def weighted_nnls_fit(spectrum, expected_matrix, constant_weight=10):
```
Comment (tacaswell) [inline skxray/core/fitting/xrf_model.py]: Removing this function needs an API change entry.

Comment (licode) [inline skxray/core/fitting/xrf_model.py]: Done

File: skxray/core/fitting/tests/test_xrf_fit.py [hunk 2]
```diff
@@ -142,15 +142,28 @@ def test_pre_fit():
...
+    # fit without weights
+    x, y_total, area_v = linear_spectrum_fitting(x0, y0, param, weights=None)
     for v in item_list:
         assert_true(v in y_total)
-
-    # no weight pre fit
-    x, y_total, area_v = linear_spectrum_fitting(x0, y0, param, constant_weight=None)
+    sum1 = np.zeros_like(x)
+    for k, v in six.iteritems(y_total):
+        sum1 += v
```
Comment (ericdill) [inline skxray/core/fitting/tests/test_xrf_fit.py]: Is this equivalent to lines 149-151?: `sum1 = np.sum(six.itervalues(y_total))`

File: skxray/core/fitting/xrf_model.py [hunk 3]
```diff
@@ -1064,72 +1064,39 @@ def nnls_fit(spectrum, expected_matrix):
...
-    a = np.transpose(np.multiply(np.transpose(standard), np.sqrt(weights)))
-    b = np.multiply(experiments, np.sqrt(weights))
-
-    [results, residue] = nnls(a, b)
+    if weights is None:
+        [results, residue] = nnls(expected_matrix, spectrum)
+    else:
+        a = np.transpose(np.multiply(np.transpose(expected_matrix), np.sqrt(weights)))
+        b = np.multiply(spectrum, np.sqrt(weights))
+        [results, residue] = nnls(a, b)
```
Comment (ericdill) [inline skxray/core/fitting/xrf_model.py]: I find this code (though identical in function to yours) easier to follow
```python
    if weights is not None:
        expected_matrix = np.transpose(np.multiply(np.transpose(expected_matrix), np.sqrt(weights)))
        spectrum = np.multiply(spectrum, np.sqrt(weights))
   return nnls(expected_matrix, spectrum)
```

File: skxray/core/fitting/xrf_model.py [hunk 4]
```diff
@@ -1143,10 +1110,8 @@ def linear_spectrum_fitting(x, y, params,
             e.g., ['Na_K', Mg_K', 'Pt_M'] refers to the
```
Comment (ericdill) [inline skxray/core/fitting/xrf_model.py]: I know that this is not part of this PR, but can you correct this docstring?

`option` -> `optional`

Can you also describe what the default behavior is going to be because this is an optional parameter?

File: skxray/core/fitting/tests/test_xrf_fit.py (hunk 2 above)
Comment (licode) [inline skxray/core/fitting/tests/test_xrf_fit.py]: yes, calculate errors for fitting with and without weights.

File: skxray/core/fitting/xrf_model.py (hunk 3 above)
Comment (licode) [inline skxray/core/fitting/xrf_model.py]: neat.

**Comment by licode** [comment]
Done.

File: skxray/core/fitting/xrf_model.py (hunk 4 above)
Comment (ericdill) [inline skxray/core/fitting/xrf_model.py]: Great, thanks

**Comment by ericdill** [comment]
Looks good.
